source/screens/featureModalScreens/FeatureModalNavigator.tsx

```diff
@@ -34,21 +42,31 @@ const FeatureModalNavigator = ({ navigation, route }: Props): JSX.Element => {
   const { startScreen = ModalScreen.Features } = route?.params || {};
 
   const [screenIndex, setScreenIndex] = useState<ModalScreen>(startScreen);
+  const [modalScreenParams, setModalScreenParams] = useState({});
```

Comment:
I was thinking if we could change the states a little bit. Instead of dividing for example when navigating inside the modal into multiple states like screenIndex and modalScreenParams, we could make an object out of that instead.

Like:

```js
const [modalScreen, setModalScreen] = useState({ screen: startScreen, props: {}}); // Modal screens
const [nextRoute, setNextRoute] = useState({ screen: '', props: {}}); // Navigator routes
```
This names are only suggestions and can of course be changed to more suitable once.

What do you think?